File: common/src/main/java/com/ssblur/scriptor/helpers/generators/DebugGenerator.java
package com.ssblur.scriptor.helpers.generators;

import com.google.gson.JsonObject;
import org.jetbrains.annotations.Nullable;

public class DebugGenerator extends TokenGenerator {

  public DebugGenerator(JsonObject object) {}

  protected DebugGenerator() {}

  public boolean canBeDefault() { return true; }

  @Override
  public String generateToken(String key, @Nullable JsonObject parameters) {
    return key;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/helpers/generators/DebugGenerator.kt
package com.ssblur.scriptor.helpers.generators

import com.google.gson.JsonObject

class DebugGenerator(@Suppress("unused_parameter") obj: JsonObject?): TokenGenerator() {
  override fun canBeDefault(): Boolean {
    return true
  }

  override fun generateToken(key: String, parameters: JsonObject?): String {
    return key
  }
}
